paper-submenu.html

```diff
@@ -63,137 +63,167 @@
-         * Toggles the submenu content when the trigger is tapped.
-         */
-        _openedChanged: function(opened, oldOpened) {
-          if (opened) {
-            this.fire('paper-submenu-open');
-          } else if (oldOpened != null) {
-            this.fire('paper-submenu-close');
-          }
-        },
+      behaviors: [
+        Polymer.IronControlState
+      ],
 
-        /**
-         * A handler that is called when `iron-activate` is fired.
-         *
-         * @param {CustomEvent} event An `iron-activate` event.
-         */
-        _onParentIronActivate: function(event) {
-          if (Polymer.Gestures.findOriginalTarget(event) !== this.__parent) {
-            return;
-          }
+      listeners: {
+        'focus': '_onFocus'
+      },
+
+      get __parent() {
+        return Polymer.dom(this).parentNode;
+      },
+
+      get __trigger() {
+        return Polymer.dom(this.$.trigger).getDistributedNodes()[0];
+      },
+
+      get __content() {
+        return Polymer.dom(this.$.content).getDistributedNodes()[0];
+      },
 
+      attached: function() {
+        this.listen(this.__parent, 'iron-activate', '_onParentIronActivate');
+      },
+
+      dettached: function() {
+        this.unlisten(this.__parent, 'iron-activate', '_onParentIronActivate');
+      },
+
+      /**
+       * Expand the submenu content.
+       */
+      open: function() {
+        if (!this.disabled && !this._active) {
+          this.$.collapse.show();
+          this._active = true;
+          this.__trigger && this.__trigger.classList.add('iron-selected');
+          this.__content && this.__content.focus();
+        }
+      },
+
+      /**
+       * Collapse the submenu content.
+       */
+      close: function() {
+        if (this._active) {
+          this.$.collapse.hide();
+          this._active = false;
+          this.__trigger && this.__trigger.classList.remove('iron-selected');
+        }
+      },
+
+      /**
+       * Toggle the submenu.
+       */
+      toggle: function() {
+        if (this._active) {
+          this.close();
+        } else {
+          this.open();
+        }
+      },
+
+      /**
+       * A handler that is called when the trigger is tapped.
+       */
+      _onTap: function(e) {
+        if (!this.disabled) {
+          this.toggle();
+        }
+      },
+
+      /**
+       * Toggles the submenu content when the trigger is tapped.
+       */
+      _openedChanged: function(opened, oldOpened) {
+        if (opened) {
+          this.fire('paper-submenu-open');
+        } else if (oldOpened != null) {
+          this.fire('paper-submenu-close');
+        }
+      },
+
+      /**
+       * A handler that is called when `iron-activate` is fired.
+       *
+       * @param {CustomEvent} event An `iron-activate` event.
+       */
+      _onParentIronActivate: function(event) {
+        var parent = this.__parent;
+        if (Polymer.Gestures.findOriginalTarget(event) === parent) {
```

Comment:
It doesn't seem like that is the case: https://github.com/Polymer/polymer/blob/42004df48ca8324bb2b98ca2a4f68393faeb526a/src/standard/gestures.html#L195-L201  cc @azakus for confirmation.